### PR TITLE
Add Cryptofuzz to our CI

### PR DESCRIPTION
**Pull request by andrewhop**
### Issues:
Resolves CryptoAlg-650

### Description of changes: 
This change adds [Cryptofuzz](https://github.com/guidovranken/cryptofuzz/) to our CI:
* A Dockerfile that can be used to build Cryptofuzz modules and Cryptofuzz itself
* CodeBuild batch config to run cryptofuzz
* A bash script to build AWS-LC and run Cryptofuzz
* A seed corpus generated from building the three modules into Cryptofuzz and running generate_corpus

### Call-outs:
This PR also refactors the fuzz test logic and puts the majority of it in the common_fuzz.sh file so it can be reused. The only real difference is how the two fuzz tests are built.

### Testing:
Tested building and running the existing fuzz tests and cryptofuzz locally. Also have it deployed to my AWS account. I expect this PR will fail because the two new Docker containers don't exist yet.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/ci/docker_images/build_cryptofuzz_modules.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  echo "export ${1}=\"${2}\"" >> "${FUZZ_ROOT}/fuzz_env.sh"
+}
+# Recommended flags from https://github.com/guidovranken/cryptofuzz/blob/master/docs/building.md
+export CFLAGS="-fsanitize=address,undefined,fuzzer-no-link -O2 -g"
+export CXXFLAGS="-fsanitize=address,undefined,fuzzer-no-link -D_GLIBCXX_DEBUG -O2 -g"
+
+# Setup base of Cryptofuzz
+cd "$FUZZ_ROOT"
+MODULES_ROOT="${FUZZ_ROOT}/modules"
+#git clone --depth 1 https://github.com/guidovranken/cryptofuzz.git
```
Comment (andrew-kaufman) [inline tests/ci/docker_images/build_cryptofuzz_modules.sh:15]: Is this still necessary?

Comment (andrewhop) [reply tests/ci/docker_images/build_cryptofuzz_modules.sh:15]: Eventually we want to clone the tip of Cryptofuzz, right now we're incompatible with the latest version until #129 is merged in. In the meantime for testing I added the checkout a specific commit below that is compatible. I'll add a note to remove it, or just do it if we get #129 merged in today.

File: tests/ci/docker_images/linux-aarch/ubuntu-20.04_cryptofuzz/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+FROM ubuntu-20.04-aarch:clang-10x
+
+SHELL ["/bin/bash", "-c"]
+
+RUN set -ex && \
+    apt-get update && \
+    apt-get -y --no-install-recommends upgrade && \
+    apt-get -y --no-install-recommends install \
+    make \
+    libboost-all-dev \
```
Comment (bryce-shang) [inline tests/ci/docker_images/linux-aarch/ubuntu-20.04_cryptofuzz/Dockerfile:13]: Is boost a dependency of cryptoFuzz?

Comment (andrewhop) [reply tests/ci/docker_images/linux-aarch/ubuntu-20.04_cryptofuzz/Dockerfile:13]: Yeah, cryptofuzz and botan.

**Review by bryce-shang — APPROVED** [review]

**Review by andrew-kaufman — APPROVED** [review]

**Pull request merged**